addons/crm/models/crm_lead.py

```diff
@@ -650,6 +723,20 @@ def _compute_is_partner_visible(self):
         for lead in self:
             lead.is_partner_visible = bool(lead.type == 'opportunity' or lead.partner_id or is_debug_mode)
 
+    @api.onchange('contact_partner_id')
+    def _onchange_contact_partner_id(self):
```

Comment:
Independently on how it is coded, should probably have unit test using the Form tool, isn't it ?